middleware/ordered_group_test.go

```diff
@@ -190,3 +190,39 @@ func TestOrderedIDsGetOrder(t *testing.T) {
 		}
 	}
 }
+
+func TestOrderedIDsSlots(t *testing.T) {
+	o := newOrderedIDs()
+
+	noError(t, o.AddSlot("first", After))
+	noError(t, o.AddSlot("second", After))
+	noError(t, o.InsertSlot("fourth", "second", After))
+	noError(t, o.Insert(mockIder("third"), "second", After))
+
+	expectIDs := []string{"first", "second", "third", "fourth"}
+	if e, a := expectIDs, o.List(); !reflect.DeepEqual(e, a) {
+		t.Errorf("expect %v order, got %v", e, a)
+	}
+
+	noError(t, o.Add(mockIder("second"), After))
```

Comment:
It would be good to have a test case that covers adding a `ider` with the same `id` multiple times, with subsequent calls failing.